### PR TITLE
BUILD: Fixing make docs

### PR DESCRIPTION
**Pull request by manjugv**
## What
Make docs depends on UCX

## Why ?
No need to depend on UCX, when requesting only to build the doxygen spec

## How ?
Disable check for UCX

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -117,30 +111,45 @@ AS_IF([test "x$with_docs_only" = xyes],
...
+     else
+	    AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_INFO], [Highest log level])
+	    CFLAGS="$CFLAGS -O3 -DNDEBUG"
+     fi
+
+     CHECK_UCX
+        AC_MSG_RESULT([UCX support: $ucx_happy])
+        if test $ucx_happy != "yes"; then
+            AC_MSG_ERROR([UCX is not available])
+        fi
```
Comment (Sergei-Lebedev) [inline configure.ac:148]: alignment, should be on the same vertical line as CHECK_UCX

**Review by Sergei-Lebedev — APPROVED** [review]

**Review by vspetrov — APPROVED** [review]

File: configure.ac [hunk 2]
```diff
@@ -117,30 +111,45 @@ AS_IF([test "x$with_docs_only" = xyes],
...
-AC_MSG_RESULT([UCX support: $ucx_happy])
-if test $ucx_happy != "yes"; then
-   AC_MSG_ERROR([UCX is not available])
-fi
+     if test $enable_debug = yes; then
+	    AC_DEFINE([ENABLE_DEBUG], [1], [Enable debugging code])
+	    CFLAGS="$CFLAGS -O0 -g3"
+	    AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_TRACE_POLL], [Highest log level])
+     else
+	    AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_INFO], [Highest log level])
```
Comment (alex--m) [inline configure.ac:140]: Probably not the highest log level

File: configure.ac [hunk 3]
```diff
@@ -117,30 +111,45 @@ AS_IF([test "x$with_docs_only" = xyes],
      UCC_DX_ENABLE_CHECK([man])
      UCC_DX_ENABLE_CHECK([pdf])
      AM_CONDITIONAL([DOCS_ONLY], [true])
+     AM_CONDITIONAL([HAVE_GTEST], [false])
+     AM_CONDITIONAL([HAVE_CXX11], [false])
+     AM_CONDITIONAL([HAVE_GNUXX11], [false])
+     AM_CONDITIONAL([HAVE_GLIBCXX_NOTHROW], [false])
+     AM_CONDITIONAL([HAVE_AARCH64_THUNDERX2], [false])
```
Comment (alex--m) [inline configure.ac:118]: Do we use these arch-specific definitions today?

Comment (manjugv) [reply configure.ac:118]: Yep, it is coming from config/m4/compiler.m4. It is because we are starting to use UCS.

File: configure.ac (hunk 2 above)
Comment (manjugv) [reply configure.ac:140]: I’m not sure what you mean here. I guess when in debug mode MAX_LOG_LEVEL is the highest level and in non-debug mode, LOG_LEVEL_INFO is the highest level. Is the description an issue, or the value? 
(Btw, this PR is not adding this line, just moving the line around. @vspetrov  can fill in better.)

Comment (alex--m) [reply configure.ac:140]: Ok, I initially thought that was a copy-paste mistake, saying that INFO is the highest log level possible (which it isn't)...

**Review by alex--m — APPROVED** [review]

**Pull request merged**